scrabble.js

```diff
@@ -1,14 +1,150 @@
+function Exception(message) {
+  this.message = message;
+  this.name = 'Exception';
+}
+
 const Scrabble = {
-  score: function(word) {
-    // TODO: implement score
-  }
+  scoreShart:  {
+    A: 1,
+    B: 3,
+    C: 3,
+    D: 2,
+    E: 1,
+    F: 4,
+    G: 2,
+    H: 4,
+    I: 1,
+    J: 8,
+    K: 5,
+    L: 1,
+    M: 3,
+    N: 1,
+    O: 1,
+    P: 3,
+    Q: 10,
+    R: 1,
+    S: 1,
+    T: 1,
+    U: 1,
+    V: 4,
+    W: 4,
+    X: 8,
+    Y: 4,
+    Z: 10
+  },
+  validWord(word) {
+    word = word.toUpperCase();
+    for (let letter of word) {
+      if (!Scrabble.scoreShart.hasOwnProperty(letter)) {
+        throw new Exception('InvalidInput');
+      }
+    }
+    return true;
+  },
+
+  score(word) {
+    let score = 0;
+    if (Scrabble.validWord(word)){
+      word = word.toUpperCase();
+      for (let letter of word) {
+        score += Scrabble.scoreShart[letter];
+      }
+    }
+
+    if (word.length ===7){
+      score+= 50;
+    } else if (word.length > 7) {
+      throw new Exception('The size of the word should be at most 7');
+    } else if (word.length === 0 ){
+      throw new Exception('The word should have length bigger than 0');
+    }
+    return score;
+  }, //close the score function
+
 
-  // TODO: add the highestScoreFrom method
+  highestScoreFrom(arrayOfWords) {
 
+    let all_scores = [];
+    for(let word of arrayOfWords) {
+      all_scores.push(Scrabble.score(word));
+    };
+    let maximo = Math.max.apply(null, all_scores);
```

Comment:
Using `apply` on a function shouldn't be necessary here - [it solves a very specific problem](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply), that I don't think you have.

Instead, call `Math.max` directly: `let maximo = Math.max(all_scores)`.